include/swift/AST/AutoDiff.h

```diff
@@ -293,23 +293,32 @@ class AutoDiffIndexSubset : public llvm::FoldingSetNode {
   AutoDiffIndexSubset &operator=(const AutoDiffIndexSubset &) = delete;
 
   // Defined in ASTContext.h.
+  static AutoDiffIndexSubset *get(ASTContext &ctx,
+                                  const SmallBitVector &indices);
+
   static AutoDiffIndexSubset *get(ASTContext &ctx,
                                   unsigned capacity,
-                                  ArrayRef<unsigned> indices);
+                                  ArrayRef<unsigned> indices) {
+    SmallBitVector indicesBitVec(capacity, false);
+    for (auto index : indices)
+      indicesBitVec.set(index);
+    return AutoDiffIndexSubset::get(ctx, indicesBitVec);
+  }
 
   static AutoDiffIndexSubset *getDefault(ASTContext &ctx,
                                          unsigned capacity,
                                          bool includeAll = false) {
-    if (includeAll)
-      return getFromRange(ctx, capacity, IntRange<>(capacity));
-    return get(ctx, capacity, {});
+    return get(ctx, SmallBitVector(capacity, includeAll));
   }
 
   static AutoDiffIndexSubset *getFromRange(ASTContext &ctx,
                                            unsigned capacity,
-                                           IntRange<> range) {
-    return get(ctx, capacity,
-               SmallVector<unsigned, 8>(range.begin(), range.end()));
+                                           unsigned start, unsigned end) {
```

Comment:
I'm thinking a lot about style after my other PR about complex numbers 😝, and was wondering if this would be better:
```suggestion
                                           unsigned start, 
                                           unsigned end) {
```

Reply:
Compared to our Swift code base, the C++ style is a bit arbitrary in this repo. My preference has been putting as many parameters as possible on one line, so it would be

```cpp
  static AutoDiffIndexSubset *getFromRange(ASTContext &ctx, unsigned capacity,
                                           unsigned start, unsigned end)
```

I'll switch to this in a later patch since CI has started.

Reply:
Actually I made more implementation changes, so I made the formatting fix as well https://github.com/apple/swift/pull/25058/commits/72ecfff72ba5fdc2fd529adf1d097ad789edf2ae.